manticore/utils/emulate.py

```diff
@@ -1,92 +1,189 @@
 import logging
+import time
 
-from ..native.memory import MemoryException
-
-from .helpers import issymbolic
+from capstone import *
 ######################################################################
 # Abstract classes for capstone/unicorn based cpus
 # no emulator by default
 from unicorn import *
-from unicorn.x86_const import *
 from unicorn.arm_const import *
+from unicorn.x86_const import *
 
-from capstone import *
+from .helpers import issymbolic
+from ..core.smtlib import Operators, solver
+from ..native.memory import MemoryException
 
 logger = logging.getLogger(__name__)
-
-
-class UnicornEmulator:
-    '''
-    Helper class to emulate a single instruction via Unicorn.
-    '''
-
+logger.setLevel(logging.INFO)
+
+
+def convert_permissions(m_perms):
+    """
+    Converts a Manticore permission string into a Unicorn permission
+    :param m_perms: Manticore perm string ('rwx')
+    :return: Unicorn Permissions
+    """
+    permissions = UC_PROT_NONE
+    if 'r' in m_perms:
+        permissions |= UC_PROT_READ
+    if 'w' in m_perms:
+        permissions |= UC_PROT_WRITE
+    if 'x' in m_perms:
+        permissions |= UC_PROT_EXEC
+    return permissions
+
+
+# https://stackoverflow.com/a/1094933
+def sizeof_fmt(num, suffix='B'):
+    for unit in ' KMGTPEZ':
+        if abs(num) < 1024.0:
+            return "%3.1f%s%s" % (num, unit if unit != ' ' else '', suffix)
+        num /= 1024.0
+    return "%.1f%s%s" % (num, 'Y', suffix)
+
+
+class ConcreteUnicornEmulator(object):
```

Comment:
Does it make sense to inherit from base emulator?

Also, py3 doesn't require to inherit from object.
```suggestion
class ConcreteUnicornEmulator:
```